File: backend/src/graphql/objects/Institution.ts
import { Field, ID, ObjectType } from "type-graphql";
import { InstitutionConnection } from "./InstitutionConnection";

@ObjectType()
export class Institution {
  @Field(_ => ID)
  id: string;

  @Field()
  externalId: string;

  @Field()
  name: string;

  @Field({nullable: true})
  logo?: string;

  @Field()
  countries: string[];

  @Field(_ => [InstitutionConnection])
  connections: InstitutionConnection[];
}

File: App.Backend/src/GraphQL/Type/Institution.cs
using GraphQL.AspNet.Attributes;

namespace App.Backend.GraphQL.Type;

public class Institution
{
    public Guid Id { get; init; }

    [GraphField(TypeExpression = "Type!")]
    public string ExternalId { get; init; } = null!;

    [GraphField(TypeExpression = "Type!")]
    public string Name { get; init; } = null!;

    public Uri? Logo { get; init; }

    [GraphField(TypeExpression = "Type!")]
    public string Country { get; init; } = null!;
}